scripts/pims2fits.py

```diff
@@ -11,9 +11,100 @@
 from datetime import datetime, timedelta
 
 from lsl.common.mcs import mjdmpm_to_datetime
+from lsl.common.paths import DATA as dataPath
+from lsl.common.stations import lwasv as lsllwasv
+from lsl.common.stations import lwana as lsllwana
+
+from astropy.coordinates import AltAz, EarthLocation, SkyCoord
+from astropy.wcs import WCS as AstroWCS
+from astropy.wcs.utils import pixel_to_skycoord
+from astropy.time import Time
+from astropy.io import fits as astrofits
 
 from lsl_toolkits.PasiImage import PasiImageDB
 
+def calcbeamprops(az,alt,header,freq):
+
+    # az and alt need to be the same shape as the image we will correct
+
+    i = 0
+    beamDict = numpy.load(os.path.join(dataPath, 'lwa1-dipole-emp.npz'))
+    polarpatterns = []
+    for beamCoeff in (beamDict['fitX'], beamDict['fitY']):
+        alphaE = numpy.polyval(beamCoeff[0,0,:],freq )
+        betaE =  numpy.polyval(beamCoeff[0,1,:],freq )
+        gammaE = numpy.polyval(beamCoeff[0,2,:],freq )
+        deltaE = numpy.polyval(beamCoeff[0,3,:],freq )
+        alphaH = numpy.polyval(beamCoeff[1,0,:],freq )
+        betaH =  numpy.polyval(beamCoeff[1,1,:],freq )
+        gammaH = numpy.polyval(beamCoeff[1,2,:],freq )
+        deltaH = numpy.polyval(beamCoeff[1,3,:],freq )
+        corrFnc = None
+
+        def compute_beam_pattern(az, alt, corr=corrFnc):
+            zaR = numpy.pi/2 - alt*numpy.pi / 180.0
+            azR = az*numpy.pi / 180.0
+
+            c = 1.0
+            if corrFnc is not None:
+                c = corrFnc(alt*numpy.pi / 180.0)
+                c = numpy.where(numpy.isfinite(c), c, 1.0)
+
+            pE = (1-(2*zaR/numpy.pi)**alphaE)*numpy.cos(zaR)**betaE + gammaE*(2*zaR/numpy.pi)*numpy.cos(zaR)**deltaE
+            pH = (1-(2*zaR/numpy.pi)**alphaH)*numpy.cos(zaR)**betaH + gammaH*(2*zaR/numpy.pi)*numpy.cos(zaR)**deltaH
+
+            return c*numpy.sqrt((pE*numpy.cos(azR))**2 + (pH*numpy.sin(azR))**2)
+        # Calculate the beam
+        pattern = compute_beam_pattern(az, alt)
+        polarpatterns.append(pattern)
+        i += 1
+    beamDict.close()
+    return polarpatterns[0], polarpatterns[1]
+
+def pbcorr(header,imSize,pScale,station):
+    sRad   = 360.0/pScale/numpy.pi / 2
+    w = AstroWCS(naxis=2)
+    w.wcs.crpix = [imSize/2 + 0.5 * ((imSize+1)%2),imSize/2  + 0.5 * ((imSize+1)%2)]
+    # 130 degrees is what is visible to the dipoles
+    w.wcs.cdelt = numpy.array([-130/imSize,130/imSize]) 
+    w.wcs.crval = [header['zenithRA'],header['zenithDec']]
+    w.wcs.ctype = ["RA---SIN", "DEC--SIN"]
```

Comment:
Should we back port any of the new additions to `lsl_toolkits.OrvilleImage`?  The two I'm thinking about are `read_all()` and the new `wcs` module.